### PR TITLE
ALTAPPS-1059: Android topics search

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[ALTAPPS-1059](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1059)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Adds search screen on Android

### CODE REVIEW

**Comment by ivan-magda** [comment]
@XanderZhu I think that it's better to push the StepScreen into the navigation stack and add the ability to come back to search results like on iOS

https://github.com/hyperskill/mobile-app/assets/8586063/fc6a7af0-8576-4d0b-bb14-0c0f43fca2e4

**Comment by XanderZhu** [comment]
@ivan-magda 
I decided to hide the search screen after navigation to the step screen because there was an animation problem while navigating back. To avoid animation problems and have a clear navigation flow I removed animations.

**Comment by ivan-magda** [comment]
@XanderZhu back button not working

Screenshot_20231215_190321

File: androidHyperskillApp/src/main/res/values/themes.xml [hunk 1]
```diff
@@ -114,8 +114,9 @@
         <item name="android:windowNoTitle">true</item>
         <item name="android:windowIsFloating">false</item>
         <item name="android:windowBackground">@color/color_background</item>
-        <item name="android:windowEnterAnimation">@anim/slide_in_from_bottom</item>
-        <item name="android:windowExitAnimation">@anim/slide_out_to_bottom</item>
+        <!--<item name="android:windowEnterAnimation">@anim/slide_in_from_bottom</item>
+        <item name="android:windowExitAnimation">@anim/slide_out_to_bottom</item>-->
```
Comment (ivan-magda) [inline androidHyperskillApp/src/main/res/values/themes.xml:118]: If we don't need this commented code, let's delete it

**Review by ivan-magda — APPROVED** [review]

**Pull request merged**